File: b_module_discover/src/main/java/com/mooc/discover/adapter/TabAdapter.java
package com.mooc.discover.adapter;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentStatePagerAdapter;

import java.util.ArrayList;

public class TabAdapter extends FragmentStatePagerAdapter {

    private ArrayList<String> titleList;
    private ArrayList<Fragment> fragments;

    public TabAdapter(FragmentManager fragmentManager, ArrayList<String> titleList, ArrayList<Fragment> fragments) {
        super(fragmentManager);
        this.titleList = titleList;
        this.fragments = fragments;
    }

    @Override
    public Fragment getItem(int position) {
        return fragments.get(position);
    }


    @Override
    public int getItemPosition(@NonNull Object object) {
        return POSITION_NONE;
    }

    @Override
    public CharSequence getPageTitle(int position) {
        return titleList.get(position);
    }


    @Override
    public int getCount() {
        return titleList.size();
    }

}

File: b_module_studyproject/src/main/java/com/mooc/studyproject/adapter/TabAdapter.kt
package com.mooc.studyproject.adapter

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter
import java.util.*

class TabAdapter(fragmentManager: FragmentManager?, private val titleList: ArrayList<String>, fragments: ArrayList<Fragment?>?) : FragmentStatePagerAdapter(fragmentManager!!) {
    private val fragments: ArrayList<Fragment?>?
    override fun getItem(position: Int): Fragment {
        return fragments?.get(position)!!
    }

    override fun getItemPosition(`object`: Any): Int {
        return super.getItemPosition(`object`)
    }

    override fun getPageTitle(position: Int): CharSequence? {
        return titleList[position]
    }

    override fun getCount(): Int {
        return titleList.size
    }

    init {
        this.fragments = fragments
    }
}